src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -142,6 +142,8 @@ private void ConnectToWorkspace(Workspace workspace)
             {
                 ResetLastParsedDocument();
 
+                _reportChangeCancellationSource = new CancellationTokenSource();
```

Comment:
Note this is being assigned when it was already assigned in the constructor, thereby leaking and never getting rid of the previous source. I suspect you want to get rid of the assignment in the constructor.

Reply:
➡️ I rewrote the first commit of this pull request to use a completely new approach for this class.